src/memory/memoryManager.ts

```diff
@@ -0,0 +1,28 @@
+import { Memory } from './memory';
+import { MaybeRelocatable, Relocatable } from './primitives/relocatable';
+import { Uint, UnsignedInteger } from './primitives/uint';
+
+export class MemorySegmentManager {
+  segmentSizes: Map<Uint, Uint>;
+  memory: Memory;
+
+  constructor() {
+    this.segmentSizes = new Map();
+    this.memory = new Memory();
+  }
+
+  addSegment(): Relocatable {
```

Comment:
what i noticed is that in memory.test.ts we access directly numSegments
should it be private?

Reply:
It's probably safer indeed, will see what I can do

Reply:
Should I make a setter then? Because we need to update these fields from the memory manager